### DISCUSSION TITLE
Add whereSlug route validator

### DISCUSSION THREAD

**Discussion by xorock** [laravel/framework#46815] [Ideas]
Currently, there are several methods in routing for input control (https://laravel.com/docs/10.x/routing#parameters-regular-expression-constraints - whereAlphaNumeric, whereNumber). A common practice is to use names in the URL, created with Str::slug e.g. /article/{slug}. 
Perhaps a cool option would be to add a validator consistent with the one generated by Str::slug method?

**Comment by shaedrich** [comment]
I think, the problem is, that while Laravel has `Str::slug()` it doesn't implement sluggifying as a whole. In order to do that, Laravel would need an implementation like Spatie's [laravel-sluggable](https://github.com/spatie/laravel-sluggable)
